src/vm/lib/vm_string.c

```diff
@@ -6,27 +6,37 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stdbool.h>
 
 int vm_strcmpl(const char *restrict s, const char *restrict s_) {
-    if (strlen(s) != strlen(s_)) {
+    int length_s = strlen(s);
```

Comment:
I think you are over-engineering. This function is too small to add abstractions like this, it isn't necessary as this will only add code that could be well read the other way.